api/spec/requests/spree/api/v2/storefront/countries_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'spec_helper'
+require 'shared_examples/api_v2/base'
+
+describe 'Taxons Spec', type: :request do
+  let!(:country) { create(:country) }
+  let!(:states)    { create_list(:state, 2, country: country) }
+  let!(:other_country) { create(:country, iso3: 'GBR') }
+
+  shared_examples 'returns valid country resource JSON' do
+    it 'returns a valid country resource JSON response' do
+      expect(response.status).to eq(200)
+
+      expect(json_response['data']).to have_type('country')
+      expect(json_response['data']).to have_relationships(:states)
+
```

Comment:
@kgorazd let's also test for proper values of all attributes: iso, iso3, iso_name, name, default, states_required, zipcode_required

Reply:
It will be ok to check only existence of all attributes in shared_examples
```
expect(json_response['data']).to have_attribute(:iso)
expect(json_response['data']).to have_attribute(:iso3)
...
```
and check later actual values in one of `#show` specs
```
expect(json_response['data']).to have_attribute(:iso).with_value(country.iso)
expect(json_response['data']).to have_attribute(:iso3).with_value(country.iso3)
...
```
?

Reply:
@kgorazd works for me! :)